src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -681,6 +681,16 @@ void CVulkanLogicalDevice::updateDescriptorSets_impl(const SUpdateDescriptorSets
             outWrite->descriptorCount = write.count;
             switch (asset::IDescriptor::GetTypeCategory(type))
             {
+                case asset::IDescriptor::EC_SAMPLER:
+                {
+                    outWrite->pImageInfo = outImageInfo;
+                    for (auto j = 0u; j < write.count; j++, outImageInfo++)
+                    {
+                        const auto& imageInfo = infos[j].info.image;
+                        // Write was validated so we can just get the sampler
+                        outImageInfo->sampler = static_cast<const CVulkanSampler*>(imageInfo.sampler.get())->getInternalObject();
```

Comment:
you need to get the `desc` as per https://github.com/Devsh-Graphics-Programming/Nabla/pull/693#discussion_r1648903341